tests/integration/envrctestcase.rs

```diff
@@ -0,0 +1,230 @@
+//! Implement a wrapper around setup and tear-down of Direnv-based test
+//! cases.
+
+use direnv::DirenvEnv;
+use lorri::{
+    build_loop::{BuildError, BuildLoop, BuildResults},
+    ops::direnv,
+    project::Project,
+    roots::Roots,
+};
+use std::collections::HashMap;
+use std::ffi::{OsStr, OsString};
+use std::fmt::Display;
+use std::fs::{create_dir, File};
+use std::io::BufWriter;
+use std::io::Write;
+use std::os::unix::ffi::OsStrExt;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempfile::{tempdir, TempDir};
+
+fn find_program<S: Display + AsRef<OsStr>>(program: S) -> PathBuf {
+    // Find programs: the environment variables in tests very likely
+    // makes PATH bogus.
+    let output = Command::new("bash")
+        .args(&["-c", "which \"$1\"", "--"])
```

Comment:
`which` is a special package that has to be installed, bash has the `type -p` builtin which does the same.